internal/charger/tplink.go

```diff
@@ -149,43 +151,58 @@ func (c *TPLink) CurrentPower() (float64, error) {
 
 // execCmd executes an TP-Link Smart Home Protocol command and provides the response
 func (c *TPLink) execCmd(cmd string) ([]byte, error) {
-	// encode command message
-	buf := bytes.NewBuffer([]byte{0, 0, 0, 0})
-	var ekey byte = 171 // initialization vector
-	for i := 0; i < len(cmd); i++ {
-		ekey = ekey ^ cmd[i]
-		_ = buf.WriteByte(ekey)
-	}
-
-	// write 4 bytes to start of buffer with command length
-	binary.BigEndian.PutUint32(buf.Bytes(), uint32(buf.Len()-4))
 
 	// open connection via TP-Link Smart Home Protocol
 	conn, err := net.DialTimeout("tcp", c.uri, 5*time.Second)
 	if err != nil {
 		return nil, err
 	}
 	defer conn.Close()
-
 	// send command
-	if _, err = buf.WriteTo(conn); err != nil {
+	if _, err = c.encode(cmd).WriteTo(conn); err != nil {
 		return nil, err
 	}
-
 	// read response
 	resp := make([]byte, 2048)
-	n, err := conn.Read(resp)
+	len, err := conn.Read(resp)
 	if err != nil {
 		return nil, err
 	}
 
+	return c.decode(resp[0:len]), nil
```

Comment:
das sicher falsch. 0:len-1?

Reply:
Wenn ich das mache, schneidet er mir das letzte Byte ab, welches die letzte schließende geschweifte Klammer des Response JSONS enthält ==> unexpected end of JSON input

Reply:
Ok, lieg ich falsch...